### ISSUE TITLE
[Release] Check list for v0.6.0

### ISSUE THREAD

**Issue by Xynnn007** [confidential-containers/community#93]
# v0.6.0

## Code freeze & release for tenant-componants (attestation-service/kbs)

Also, this release also includes KBS/AS. We might need the following steps:
- [ ] 1. Cut an attestation-service v0.6.0 and make images for AS and RVPS, if changes happened in the project. 
```
   * https://github.com/confidential-containers/attestation-service
   * Cut a release (AS/RVPS images will be automatically built triggered by release)
```

- [ ] 2. Update kbs to use the latest commit from attestation-service, cut a release and make image
```
  * https://github.com/confidential-containers/kbs/blob/main/src/api_server/Cargo.toml
  * Change the revision for the following crates (both use `v0.6.0`)
     * `as-types`
     * `attestation-service`
  * Cut a release (kbs image will be automatically built triggered by release)
```

> Note: If there are breaking changes in the communication protocol between KBS and KBC, consider replacing the image in CI with the AS/KBS image of the newly released.

## Code freeze for guest-componants (attestation-agent/ocicrypt-rs//image-rs)

- [ ] - 1. Update image-rs to use the latest commit from ocicrypt-rs
        
        * https://github.com/confidential-containers/image-rs/blob/main/Cargo.toml
          * Change the revision
          * Run `cargo update -p ocicrypt-rs`

- [ ] - 2. Update image-rs to use the latest commit from attestation-agent

        * https://github.com/confidential-containers/image-rs/blob/main/Cargo.toml
          * Change the revision
          * Run `cargo update -p attestation_agent`

- [ ] - 3. Update Enclave CC to use the latest commit from image-rs

        * https://github.com/confidential-containers/enclave-cc/blob/main/src/enclave-agent/Cargo.toml
          * Change the revision
          * Run `cargo update -p image-rs`
        Note that you can point to your own fork here, so you don't actually do changes in the other projects
        before making sure this step works as expected.

- [ ] - 4. Update Kata Containers to use the latest commit from image-rs

        * https://github.com/kata-containers/kata-containers/blob/CCv0/src/agent/Cargo.toml
          * Change the revision
          * Run `cargo update -p image-rs`
        Note that you can point to your own fork here, so you don't actually do changes in the other projects
        before making sure this step works as expected.

- [ ] - 5. Update Kata Containers to use the latest attestation-agent

        * https://github.com/kata-containers/kata-containers/blob/CCv0/versions.yaml
          * Change the version

- [ ] - 6. Update Kata Containers to use the latest td-shim

        * https://github.com/kata-containers/kata-containers/blob/CCv0/versions.yaml
          * Change the version

- [ ] - 7. Check if there are new changes in the pre install payload script

        * https://github.com/confidential-containers/operator/tree/main/install/pre-install-payload
          * The last commit there must match what's in the following files as preInstall / postUninstall image
            * Enclave CC: https://github.com/confidential-containers/operator/blob/main/config/samples/enclave-cc/base/ccruntime-enclave-cc.yaml
            * Kata Containers:
              Note that for Kata Containers, we're looking for the newTag, below the quay.io/confidential-containers/container-engine-for-cc-payload image
              * s390x: https://github.com/confidential-containers/operator/blob/main/config/samples/ccruntime/s390x/kustomization.yaml
              * x86_64: https://github.com/confidential-containers/operator/blob/main/config/samples/ccruntime/default/kustomization.yaml

- [ ] - 8. Ensure the Operator is using the latest CI builds and that the Operator tests are passsing

        * Enclave CC:
          * SIM: https://github.com/confidential-containers/operator/blob/main/config/samples/enclave-cc/sim/kustomization.yaml
          * HW: https://github.com/confidential-containers/operator/blob/main/config/samples/enclave-cc/base/ccruntime-enclave-cc.yaml
          * Note that we need the quay.io/confidential-containers/runtime-payload-ci registry and enclave-cc-{SIM,HW}-latest tags
        * Kata Containers:
          * s390x: https://github.com/confidential-containers/operator/blob/main/config/samples/ccruntime/s390x/kustomization.yaml
          * x86_64: https://github.com/confidential-containers/operator/blob/main/config/samples/ccruntime/default/kustomization.yaml
          Note that we need the quay.io/confidential-containers/runtime-payload-ci registry and kata-containers-latest tag

- [ ] - 9. Cut an ocicrypt-rs v0.6.0 release, if changes happened in the project

- [ ] - 10. Cut an attestation-agent v0.6.0, if changes happened in the project

- [ ] - 11. Cut an image-rs v0.6.0 release, using the latest release of:

        * ocicrypt-rs (redo step 1, but now using v0.6.0)
        * attestation-agent (redo step 2, but now using v0.6.0)

- [ ] - 12. Cut a td-shim v0.6.0 release, if changes happened in the project

- [ ] - 13. Update Enclave CC to use the released version of image-rs

        * redo step 3, but now using v0.6.0

- [ ] - 14. Update Kata Containers to the latest released version of:

        * image-rs (redo step 4, but now using the v0.6.0)
        * attestation-agent (redo step 5, but now using the v0.6.0)
        * td-shim (redo step 6, but now using the v0.6.0)

- [ ] - 15. Update the operator to use the images generated from the latest commit of both Kata Containers and Enclave CC

        * redo step 8, but now targetting the latest payload image generated for Kata Containers and Enclave CC

- [ ] - 16. Make sure all the operator tests are passing

- [ ] - 17. Cut an Enclave CC release

- [ ] - 20. Add a new Kata Containers tag

## Release

- [ ] - 21. Update the operator to use the release tags coming from Enclave CC and Kata Containers

        * redo step 8, but now targetting thje latest release of the payload image generated for Kata Containers eand Enclave CC

- [ ] - 22. Update the Operator version

        * https://github.com/confidential-containers/operator/blob/main/config/release/kustomization.yaml#L7

- [ ] - 23. Cut an operator release

- [ ] - 24. Make sure to update the release notes

        * https://github.com/confidential-containers/documentation/tree/main/releases/v0.6.0.md
        
- [ ] - 25. Poke Jens Freimann (jfreiman@redhat.com) to update the release to the OperatorHub

**Comment by Xynnn007** [comment]
Change the order slightly, because CI may use the new image that has just been released

**Comment by zvonkok** [comment]
On which version of Kata will 0.6.0 be based? I also want to make sure we have https://github.com/kata-containers/kata-containers/pull/6699 in the 0.6.0

**Comment by fitzthum** [comment]
@stevenhorsman This checklist doesn't contain anything regarding peer pods. What is the process for the release there and are there things we should be adding to the checklist?

**Comment by stevenhorsman** [comment]
> @stevenhorsman This checklist doesn't contain anything regarding peer pods. What is the process for the release there and are there things we should be adding to the checklist?

Yeah, I'll add an extra section here for it. Thanks for the reminder

**Comment by fitzthum** [comment]
@zvonkok Release v0.6.0 will use a bundle built from the CCv0 branch of Kata (see step 15). So changes for this release may need to get backported to CCv0.

This checklist is really for tasks that facilitate the release itself. If there are more things to get in before the freeze starts, then your best bet is to get them merged before EOD today. Otherwise it's a question of whether to delay the freeze, which would need to be a wider discussion.

Hopefully the next release will be based on main, so if things don't make it into this one it should be simpler next time around.

**Comment by sprt** [comment]
Can't tag myself but I'll update the Kata Containers dependencies.

**Comment by Xynnn007** [comment]
> @Xynnn007 it'll be good if KBS image is pushed to quay.io/confidential-containers registry like the other images.

Yes @bpradipt , the release CI will do this, after release guest components. 

The reason is that today I found that kbs refers to AS' code https://github.com/confidential-containers/kbs/blob/main/tools/client/Cargo.toml#L16, so my idea is to cut release for the guest components and then change the rev inside that Cargo.toml.

BTW, as currently only tdx has been using kbs in CI afaik, Arron has helped update the tdx ci image already. Thus we do not need to worry about the CI in kata

**Comment by mythi** [comment]
> For enclave-cc, we are not pointing to the latest preInstall image (see [here](https://github.com/confidential-containers/operator/blob/fddce46d98e96ede3ebd1274d2a696eb8ec4d876/config/samples/enclave-cc/base/ccruntime-enclave-cc.yaml#L88)). This is also not the most recent hash. Is this intentional @mythi or should this also be updated to point to latest?

It's not intentional, thanks for noticing! I had missed that other `CcRuntime`s had moved to kustomization for these too. However, it looks the containerd version has not changed since 0.5.0 which triggers the question have these images been updated?

**Comment by stevenhorsman** [comment]
> Regarding number 7, the s390 and default operator yamls both point to the `latest` tag of `container-engine-for-cc-payload`. Does that seem correct to you @stevenhorsman or should we be fixing the yaml to a particular hash for the release? We haven't updated the preInstall container in two months anyway.

Yes - I agree that we should switch it back to a proper hash for the release. I suspect that it wasn't worth my switching it to latest when I did the same for the runtime payload to be honest as we don't update that payload often enough, so I'll try and remember to not do that next time.

The other thing to note is that when I made the runtime-payload multi-arch I switched the s390x to base on default, so I don't think we need the specific s390x update any more - I updated part 8 in the checklist so remove it, but forget part 7, so apologies for any confusion and I'll update the template once the release has gone out.

I hope that helps and answers the question?

**Comment by fitzthum** [comment]
Ok, just waiting for the payload build to finish and then I will do 7 and 8.

**Comment by fitzthum** [comment]
Nvm, we have https://github.com/confidential-containers/operator/pull/215

**Comment by stevenhorsman** [comment]
@Xynnn007 - do you know if anyone is working on creating the releases of the components (steps 9-12), or is help needed there?

**Comment by Xynnn007** [comment]
@stevenhorsman 
I can do 10 & 11, while I do not have permission in AA (9) and td-shim (12). Could you help with this?

**Comment by stevenhorsman** [comment]
> AFAIK, no. I can do 10 & 11, while I do not have permission in AA (9) and td-shim (12). Could you help with this?

Sure - I can help with which ever numbers you like and @fitzthum has a github team for release-champions I believe that would give you access to create the tags.

**Comment by stevenhorsman** [comment]
Just as an update on 12 - td-shim has had two commits added since we tested it in the release candidate. I'm hoping that they aren't required, but I've messaged Jiewen to confirm before we cut the release - but if anyone else knows before we get the update that would help us move onto the next steps quicker.

**Comment by stevenhorsman** [comment]
I've done up to step 13, which has the PR https://github.com/confidential-containers/enclave-cc/pull/168 in review.
Over to @sprt for step 14 now :)

**Comment by sprt** [comment]
@Xynnn007 PR for step 14: https://github.com/kata-containers/kata-containers/pull/7037

**Comment by Xynnn007** [comment]
> Step 16 tests all passed, so I think we are ready to release enclave-cc & tag kata-containers. Are there already people signed up to look at those tasks?


Let me cut enclave-cc

**Comment by fitzthum** [comment]
Who has permissions to make tag for Kata?

**Comment by stevenhorsman** [comment]
I think the Kata tag has worked correctly: https://github.com/kata-containers/kata-containers/releases/tag/CC-0.6.0

**Comment by stevenhorsman** [comment]
BTW - I'm working on 19 now

**Comment by fitzthum** [comment]
We may have a minor issue with enclave-cc release workflow that we need https://github.com/confidential-containers/enclave-cc/pull/170 to fix

**Comment by fitzthum** [comment]
@jensfr can you take a look at step 24?

**Comment by fitzthum** [comment]
I have created an issue with some reflections https://github.com/confidential-containers/community/issues/95

**Comment by stevenhorsman** [comment]
Hey folks, I just want to clarify/tie up a loose end on the operator. There was a bug in the operator hub install that got fixed yesterday (with https://github.com/confidential-containers/operator/pull/214), after the v0.6.0 release was cut. Do we want to cut a v0.6.1 which is just v0.6.0 + that fix before we get Jens to update operator hub?

**Comment by fitzthum** [comment]
I think that is a good idea. No point in releasing v0.6.0 on the operator hub.

**Comment by fitzthum** [comment]
Nobody is signed up. Feel free to make the release.

**Comment by stevenhorsman** [comment]
> Nobody is signed up. Feel free to make the release.

Done: @jensfr - https://github.com/confidential-containers/operator/releases/tag/v0.6.1 should be good for operator hub when you are ready

**Comment by Xynnn007** [comment]
Can we close this issue now?
